src/compiler/diagnosticMessages.json

```diff
@@ -2469,5 +2469,9 @@
     "A type assertion expression is not allowed in the left-hand side of an exponentiation expression. Consider enclosing the expression in parentheses.": {
       "category": "Error",
       "code": 17007
+    },
+    "The body of an 'if' statement cannot be the empty statement.": {
+      "category": "Warning",

```

Comment:
Just make this an error the same as any other (probably in the 1xxx category); we don't need to introduce a new concept of a "warning" right now